src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs

```diff
@@ -352,13 +352,18 @@ public IReadOnlyList<Token> LexSource(string text, Flags flags = Flags.None)
         {
             Contracts.AssertValue(text);
 
-            var tokens = new List<Token>();
+            List<Token> tokens;
             StringBuilder sb = null;
 
             try
             {
                 // This StringBuilder is used by the Lexer as a temporary storage for tokenized characters.
-                sb = StringBuilderCache.Acquire(Math.Min(text.Length, DesiredStringBuilderSize));
+                sb = StringBuilderCache.Acquire(Math.Min(text.Length, DesiredStringBuilderSize));
+
+                if (TryParseTextFirstEmptyString(text, flags, out tokens))
+                {
+                    return tokens;
+                }
```

Comment:
This can be moved to before the creation (acquiring) of the StringBuilder - since it's not used.

We can also move it outside the `try` block, as this should not throw.